### DISCUSSION TITLE
NER with pre-tokenized sentences

### DISCUSSION THREAD

**Discussion by ophelielacroix** [peleiden/daLUKE#113] [Q&A] [answered]
Hello, 

is there a way to run the NER model on pre-tokenized sentences?

**Comment by sorenmulli** [comment]
Hello Ophelia

Thank you for your interest in our model.
Sorry about the super slow answer; we did not get notifications from discussion posts.

No, the current API performs tokenization itself and finally returns predictions on the word-level (e.g. space-separated).
I can add this feature, I just want to hear how this should work: The function should take a list of tokens and then return its' NER classifications (B-ORG, I-ORG, ...) on the token level and not the word level?

**Comment by ophelielacroix** [comment]
Hi, 
thanks for the answer. Yes, I think it would take a list of tokens as input and return a list of NER tags of the same length corresponding to the tokens.

**Comment by sorenmulli** [comment, marked as answer]
Yes, that would be possible in current daluke version:

If the input is 
```py
tokenized_sentences = [["Example", "sentence", "one"], ["Example", "sentence", "number", "two"]]
```
then you would get the desired result using
```py
from daluke import AutoNERDaLUKE, predict_ner

model = AutoNERDaLUKE()
input_strings = [" ".join(s) for s in tokenized_sentences]
token_ner = predict_ner(input_strings, model) # <- batch size can also be controlled here
print(token_ner) # Gives: [["O", "O", "O"], ["O", "O", "O", "O"]]
```

Please let me know if this does not work, or if I misunderstood the task :)

**Comment by ophelielacroix** [comment]
Thanks for looking at that :)
Then, would that ensure to get a list of output tags that has the same length as the list of input tokens for each sentence ?

**Reply by sorenmulli** [reply]
Yes, using this approach (`[" ".join(s) for s in tokenized_sentences]`), the lists should have the same length!

**Reply by ophelielacroix** [reply]
Great! Thanks :)
